lib/cfe_internal_hub.cf

```diff
@@ -204,8 +204,8 @@ bundle agent cfe_internal_database_cleanup_consumer_status (row_count)
   vars:
 
       "status_table_name" -> { "ENT-4331" }
-        string => ifelse( eval( "$(sys.cf_version) < 3.12.1", class, infix ), "__status",
-                          "status"),
+        string => ifelse( eval( "$(sys.cf_version) < 3.12.2", class, infix ), "status",
+                          "__status"),
```

Comment:
Would be better to use `cf_version_major/minor/patch`, otherwise the comparison fails on pre-release builds because variables there look like this:
```
default:sys.cf_version                   3.12.2a.8f887ce                                              inventory,source=agent,attribute_name=CFEngine version
default:sys.cf_version_major             3                                                            source=agent                            
default:sys.cf_version_minor             12                                                           source=agent                            
default:sys.cf_version_patch             2                                                            source=agent
```

Reply:
I could not resolve the full matrix of comparisons trying to compare the parts separately in my head.
Maybe you have some simple logic that I have overlooked, but I did not get very far down the road before the expressions was not ledible.